defaults/host-ents.go

```diff
@@ -0,0 +1,220 @@
+ */
+func hostDevicesNoneEntitlementEnforce(profile secprofile.Profile) (secprofile.Profile, error) {
+	ociProfile, err := ociProfileConversionCheck(profile, HostDevicesNoneEntFullID)
+	if err != nil {
+		return nil, err
+	}
+
+	capsToRemove := []types.Capability{
+		CapSysAdmin,
+	}
+	ociProfile.RemoveCaps(capsToRemove...)
+
+	ociProfile.AppArmorSetup.Files.ReadOnly = append(ociProfile.AppArmorSetup.Files.ReadOnly, "/sys/**")
+	ociProfile.AppArmorSetup.Files.Denied = append(ociProfile.AppArmorSetup.Files.Denied, "/proc/kcore/**")
+
+	ociProfile.OCI.Linux.ReadonlyPaths = append(ociProfile.OCI.Linux.ReadonlyPaths, "/sys")
+	ociProfile.OCI.Linux.MaskedPaths = append(ociProfile.OCI.Linux.MaskedPaths, "/proc/kcore")
+
+	ociProfile.OCI.Mounts = defaultMobyAllowedMounts
+
+	return ociProfile, nil
+}
+
+/* Implements "host.devices.view" entitlement
+ * - Sets all custom mounts to read-only
+ */
+func hostDevicesViewEntitlementEnforce(profile secprofile.Profile) (secprofile.Profile, error) {
+	ociProfile, err := ociProfileConversionCheck(profile, HostDevicesViewEntFullID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mount := range ociProfile.OCI.Mounts {
+		mountPath := mount.Destination
+
+		if !isAllowedMount(mount) {
+			ociProfile.OCI.Linux.ReadonlyPaths = append(ociProfile.OCI.Linux.ReadonlyPaths, mountPath)
+		}
+	}
+
+	return ociProfile, nil
+}
+
+// removeReadOnlyFlagMounts removes the read-only "ro" flag from a mount object's mount options
+func removeReadOnlyFlagMount(mount specs.Mount) specs.Mount {
+	readWriteMount := mount
+	for index, option := range readWriteMount.Options {
+		if option == "ro" {
+			readWriteMount.Options = append(readWriteMount.Options[:index], readWriteMount.Options[index+1:]...)
+			break
+		}
+	}
+
+	return readWriteMount
+}
+
+// removeReadOnlyFlagMounts removes the read-only "ro" flag from mount options of a provided list of mounts
+func removeReadOnlyFlagMounts(mounts []specs.Mount) []specs.Mount {
+	readWriteMounts := make([]specs.Mount, len(mounts))
+
+	for mountIndex, mount := range mounts {
+		readWriteMounts[mountIndex] = removeReadOnlyFlagMount(mount)
+	}
+
+	return readWriteMounts
+}
+
+/* Implements "host.devices.admin" entitlement
+ * - Allowed Capabilities: CAP_SYS_ADMIN
+ * - Sets mounts as Read-Write
+ * - No paths masked to the container
+ */
+func hostDevicesAdminEntitlementEnforce(profile secprofile.Profile) (secprofile.Profile, error) {
+	ociProfile, err := ociProfileConversionCheck(profile, HostDevicesAdminEntFullID)
+	if err != nil {
+		return nil, err
+	}
+
+	// FIXME: just remove read-only flags for default mounts and leave additional mounts as is
+	ociProfile.OCI.Mounts = removeReadOnlyFlagMounts(defaultMobyAllowedMounts)
+
+	ociProfile.OCI.Linux.MaskedPaths = []string{}
+
+	capsToAdd := []types.Capability{
+		CapSysAdmin,
+	}
+	ociProfile.AddCaps(capsToAdd...)
+
+	return ociProfile, nil
+}
+
+/* Implements "host.processes.none" entitlement
+ * - Activates User Namespace
```

Comment:
Why is `.host.processes.none` related to the user namespace? Shouldn't this be pid namespace?